### PR TITLE
Update XMOJ.user.js

### PR DESCRIPTION
**Pull request by PythonSmall-Q**
add redirection

### CODE REVIEW

**Comment by PythonSmall-Q** [comment]
你把那个xmojtest改成xmoj。忘记复原了

File: XMOJ.user.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // ==UserScript==
-// @name         XMOJ
+// @name         XMOJTest
 // @version      0.1.34
 // @description  XMOJ增强脚本
-// @author       @langningchen
+// @author       @langningcheng
```
Comment (langningchen) [inline XMOJ.user.js:5]: There is a typo in my name

**Comment by PythonSmall-Q** [comment]
Okay. validate

**Comment by langningchen** [comment]
We don't use jQuery, could you please replace it?

**Comment by langningchen** [comment]
And, do you write the code by yourself or just copy from the web?

**Comment by PythonSmall-Q** [comment]
it can work with jquery

**Comment by PythonSmall-Q** [comment]
just to make sure no conflicts

File: XMOJ.user.js [hunk 2]
```diff
@@ -19,6 +19,240 @@
...
+        if (!url || replaced.has(url) || url.includes("localhost") || url.includes("127.0.0.1") || existingLinks.has(url)) {
+          continue;
+        }
+        if (url.endsWith(".") && url[url.length - 2].match(/\d/g)) {
+          url = url.substring(0, url.length - 2);
+        }
+        replaceSingleURL(url);
+        replaced.add(url);
+      }
+    }
```
Comment (langningchen) [inline XMOJ.user.js:136]: What's wrong with WeChat?

Comment (PythonSmall-Q) [reply XMOJ.user.js:136]: Where is Wehcattee

File: XMOJ.user.js [hunk 3]
```diff
@@ -19,6 +19,240 @@
...
+ * @param {String} url
+ */
+function replaceSingleURL(url) {
+  $("#js_content").html((_, html) => {
+    return html.replaceAll(url, `<a target="_blank" rel="noopener noreferrer" href="${url}">${url}</a>`);
+  });
+}
+
+/**
+ * Make urls clickable again on Weixin Media Platform.
```
Comment (langningchen) [inline XMOJ.user.js:107]: WeChat is here: "Weixin Media Platform"

File: XMOJ.user.js (hunk 2 above)
Comment (PythonSmall-Q) [reply XMOJ.user.js:136]: didn't find it

File: XMOJ.user.js (hunk 3 above)
Comment (PythonSmall-Q) [reply XMOJ.user.js:107]: oh, it's just if GLS send a 公众号文章

Comment (PythonSmall-Q) [reply XMOJ.user.js:107]: and linked that to xmoj, it might be xmoj.tech/posts/*

Comment (PythonSmall-Q) [reply XMOJ.user.js:107]: for some time, but now that is fixed. i forgot to delete that comment

Comment (PythonSmall-Q) [reply XMOJ.user.js:107]: plz delete that, thx

File: XMOJ.user.js [hunk 4]
```diff
@@ -116,6 +116,10 @@ GM_registerMenuCommand("重置数据", () => {
     }
 });
 
+if (location.host != "www.xmoj.tech") {
+    location.host = "www.xmoj.tech";
+}
```
Comment (langningchen) [inline XMOJ.user.js:121]: This can be good. Yours is too complex. 😆

Comment (langningchen) [reply XMOJ.user.js:121]: If you have problem, please type here instead of calling me in class.

**Comment by PythonSmall-Q** [comment]
okay, use yours

**Comment by langningchen** [comment]
I will close this pr. Since the modify is too small, it's not necessary to make a release by merging it. I will fix #5 in the next release. Thanks for your contribution. 🎆

**Comment by langningchen** [comment]
I will close this pr. Since the modify is too small, it's not necessary to make a release by merging it. I will fix #5 in the next release. Thanks for your contributing.

**Pull request closed without merging**